packages/template-retail-react-app/app/pages/checkout/index.test.js

```diff
@@ -299,6 +299,16 @@ test('Can proceed through checkout steps as guest', async () => {
     await user.click(screen.getByText(/2 items in cart/i))
     expect(await screen.findByText(/Long Sleeve Crew Neck/i)).toBeInTheDocument()
 
+    // Verify password field is reset if customer toggles login form
+    const loginToggleButton = screen.getByText(/Already have an account\? Log in/i)
+    await user.click(loginToggleButton)
+    // Provide customer email and submit
+    const passwordInput = document.querySelector('input[type="password"]')
+    await user.type(passwordInput, 'Password1!')
+
+    const checkoutAsGuestButton = screen.getByText(/Checkout as guest/i)
+    await user.click(checkoutAsGuestButton)
+
```

Comment:
It wasn't super obvious to me how this is verifying the field is reset when the form is toggled until I looked at the code in contact-info and saw that the condition for attempting a login is that the password field is not empty.

A comment here saying that `clicking Checkout as Guest clears the password field so a login is not attempted` might help?

Reply:
In addition to comments, should there be assertions validating that the state changed as expected?